### ISSUE TITLE
View Shared Tooltip/Crosshair without hovering over graph by using variable

### ISSUE THREAD

**Issue by zcase** [grafana/grafana#33952]
**What would you like to be added**:
Would like the ability to display the shared crosshair/tooltip in graphs (or other plugins) without having to hover over graphs. Would like them to be controlled by a dashboard variable.

**Why is this needed**:
Currently have a Cesium plugin and as I slide the timeline it updates the current time within the cesium plugin. I would like to hook that to a variable and as the timeline updates the current time it would also update the variable it is attached to. As the variable is updated via sliding the timeline in the cesium plugin, then in will show the tooltip/crosshair automatically (without hovering over the graphs) in the graphs that have been configured to rely on the variable.

If this is currently possible to link crosshairs/tooltips to dashboard variables, is there an existing example?

**Comment by marefr** [comment]
Would it be possible to include some screenshots to help us understand your use case better? 

> Would like them to be controlled by a dashboard variable.

Please explain how you imagine this will work. Given variable a, when I select `<value>` then what should happen?

**Comment by zcase** [comment]
![GraphanaQuestion](https://user-images.githubusercontent.com/15258806/118161978-752d9d00-b3dd-11eb-962d-4226be17b3d1.PNG)

In the above picture with the label 1 is the cesium timeline and what I'm looking to do is tie that to a dashboard variable so that when the user slides that back and forth with the cesium plugin, the dashboard variable is updated to the current time the slider is set to.

Then as this dashboard variable is being updated via the user sliding the timeline slider, it would trigger the vertical lines (that usually appear when a user hovers over the graph (the crosshair/tooltip)) to become visible and slide with the time set on the x axis of the graphs, so that the visualization within the cesium plugin match that of the current time the tooltip/crosshair is on in the graphs.

Basically I just want the user to be able to visually see the point in the graphs that the cesium timeline is referencing given the data in db the graphs are referencing have the same timeline.

**Comment by marefr** [comment]
Thanks. I'm unsure this is something we can/want to support. @grafana/grafana-frontend-platform any input?

**Comment by zcase** [comment]
I feel like this would be a very beneficial feature especially for 3rd party plugin developers. Especially if the plugin has time information related to what is being graphed like in my example above. It makes for a very smooth/automatic analysis of what is on the page rather than the manual approach of sliding the cesium timeline slider (which you have to do anyways) and then manually having to go to the graphs and find the right spot to hover over.

**Comment by ryantxu** [comment]
In 8x, this can be supported by broadcasting [DataHoverEvent](https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/events/common.ts#L12) over the EventBus

**Comment by zcase** [comment]
Is there an example I can look at from 8x of something similar being done?

**Comment by justintocco** [comment]
I'm actually looking for a very similar functionality. I'd like my custom panel to send a DataHoverEvent to other time series graphs, but really struggling to get any response from my broadcast.

The closest thing I see to building an event and broadcasting to other time series panels is [this TimeSeries/utils.ts file](https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/components/TimeSeries/utils.ts). However, this seems way too complicated to implement from scratch for a custom panel.

How much data (or what kind of format) is required for the DataHoverEvent to be acknowledged by other graphs? Currently, I am just sending a DataHoverPayload with the "point" field filled (which is the only required field), but still getting no response.

Any help on this would be greatly appreciated by me (and probably zcase too). Thanks!

**Comment by dprokop** [comment]
@justintocco do you mind sharing repository with your code? Maybe looking and tinkering with the code will allow us to help you :)

**Comment by justintocco** [comment]
Here is my [Panel Source Code](https://github.com/justintocco/FF-map/tree/main). I only included the file where I attempt to construct and broadcast a cursor event (lines 120-127).

Let me know if you have any further questions.
